tests/MongoDB.Driver.Examples/InsertDataWithEncryptedFieldsWithAutoEncryption.cs

```diff
@@ -0,0 +1,295 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using MongoDB.Bson;
+using MongoDB.Driver.Encryption;
+using Xunit;
+
+namespace MongoDB.Driver.Examples
+{
+    public class InsertDataWithEncryptedFieldsWithAutoEncryption
+    {
+        private static readonly string __connectionString = "mongodb://localhost:27017";
+        private static readonly string __sampleNameValue = "John Doe";
+        private static readonly int __sampleSsnValue = 145014000;
+
+        private static BsonDocument SampleDoc =
+            new BsonDocument
+            {
+                { "name", __sampleNameValue },
+                { "ssn", __sampleSsnValue },
+                { "bloodType", "AB-" },
+                {
+                    "medicalRecords",
+                    new BsonArray(new []
+                    {
+                        new BsonDocument("weight", 180),
+                        new BsonDocument("bloodPressure", "120/80")
+                    })
+                },
+                {
+                    "insurance",
+                    new BsonDocument
+                    {
+                        { "policyNumber", 123142 },
+                        { "provider", "MaestCare" }
+                    }
+                }
+            };
+
+        // https://gist.github.com/DmitryLukyanov/8417d1f5c30f12ffeb2be29056aff455#file-insertdatawithencryptedfields-cs
+        [Fact]
+        public void InsertDataWithEncryptedFields()
+        {
+            var keyVaultNamespace = CollectionNamespace.FromFullName("encryption.__keyVault");
+            var medicalRecordsNamespace = CollectionNamespace.FromFullName("medicalRecords.patients");
+
+            var base64KeyId = ""; // paste the generated key in Standard GuidRepresentation form here
+
+            var ssnQuery = Builders<BsonDocument>.Filter.Eq("ssn", __sampleSsnValue);
+
+            // Construct a JSON Schema
+            var schema = JsonSchemaCreator.CreateJsonSchema(base64KeyId);
+
+            // Construct an encrypted client
+            var encryptedClient = CreateEncryptedClient(
```

Comment:
// construct an auto-encrypting client
    var autoEncryptingClient = ...

The client is not encrypted... it is a client that does auto encryption.